<?xml version="1.0"?>

<launch>

<!-- TF  -->

  <node pkg="guidance" type="tf_publisher" name="tf_publisher"/>
  <node pkg="tf" type="static_transform_publisher" name="static_transform_publisher"
    args=" 0.08 0 0.05 0 0 0 M100_base_link laser 20"/>

<!-- Camera Info  -->

  <node pkg="guidance" type="camerainfo_publisher" name="camerainfo_publisher"
  args="front rear left right down"/>

<!-- Nao Locator  -->

  <node pkg="guidance" type="maptomarker_tf" name="maptomarker_tf"
  args="ar_marker_4"/>
	<node pkg="guidance" type="armarker_subscriber" name="armarker_subscriber" output="screen"/> 

<!-- Hokuyo Lidar  -->

  <node pkg="urg_node" type="urg_node" name="urg_node"
    args=" _ip_address:='192.168.1.10' "/>

<!-- lidartocloud and cloudtolasernao -->

<!--	<node pkg="guidance" type="laserscan_transform_toNao" name="laserscan_transform_toNao" output="screen"/> 
<node pkg="pointcloud_to_laserscan" type="pointcloud_to_laserscan_node" name="pointcloud_to_laserscan_node" output="screen"/> 
-->	

<!-- Hector SLAM -->

  <arg name="geotiff_map_file_path" default="$(find hector_geotiff)/maps"/>

 <include file="$(find hector_mapping)/launch/mapping_default.launch">
    <arg name="tf_map_scanmatch_transform_frame_name" value="M100_base_link"/>
    <arg name="base_frame" value="M100_base_link"/>
    <arg name="odom_frame" value="M100_base_link"/>
 </include>

  <include file="$(find hector_geotiff)/launch/geotiff_mapper.launch">
    <arg name="trajectory_source_frame_name" value="M100_base_link"/>
    <arg name="map_file_path" value="$(arg geotiff_map_file_path)"/>
  </include>
 
<!--   <node pkg="hector_exploration_node" type="exploration_planner_node" name="hector_exploration_node" output="screen">
     <rosparam file="$(find hector_exploration_node)/config/costmap.yaml" command="load" />
  </node>
 -->

<!--            MARKER DETECTION                           -->

    <arg name="marker_size" default="7" />
  <arg name="max_new_marker_error" default="0.08" />
  <arg name="max_track_error" default="0.2" />
  <arg name="cam_image_topic" default="/guidance/front/left_camera/image_raw" />
  <arg name="cam_info_topic" default="/guidance/front/left_camera/camera_info" />
  <arg name="output_frame" default="/M100_base_link" />

  <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
    <param name="marker_size"           type="double" value="$(arg marker_size)" />
    <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
    <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
    <param name="output_frame"          type="string" value="$(arg output_frame)" />

    <remap from="camera_image"  to="$(arg cam_image_topic)" />
    <remap from="camera_info"   to="$(arg cam_info_topic)" />
  </node> 


<!--	<node pkg="amcl" type="amcl" name="amcl" output="screen">
		<param name="odom_model_type" value="diff"/>
		<param name="odom_alpha5" value="0.1"/>
		<param name="transform_tolerance" value="0.2" />
		<param name="gui_publish_rate" value="1.0"/>
		<param name="laser_max_beams" value="30"/>
		<param name="min_particles" value="500"/>
		<param name="max_particles" value="5000"/>
		<param name="kld_err" value="0.05"/>
		<param name="kld_z" value="0.99"/>
		<param name="odom_alpha1" value="0.2"/>
		<param name="odom_alpha2" value="0.2"/>
	
	<param name="odom_alpha3" value="0.8"/>
		<param name="odom_alpha4" value="0.2"/>
		<param name="laser_z_hit" value="0.5"/>
		<param name="laser_z_short" value="0.05"/>
		<param name="laser_z_max" value="0.05"/>
		<param name="laser_z_rand" value="0.5"/>
		<param name="laser_sigma_hit" value="0.2"/>
		<param name="laser_lambda_short" value="0.1"/>
		<param name="laser_lambda_short" value="0.1"/>
		<param name="laser_model_type" value="likelihood_field"/>
		
		<param name="use_map_topic" value="true"/>
		<param name="laser_likelihood_max_dist" value="2.0"/>
		<param name="update_min_d" value="0.2"/>
		<param name="update_min_a" value="0.5"/>
		<param name="odom_frame_id" value="odom"/>
		<param name="resample_interval" value="1"/>
		<param name="transform_tolerance" value="0.1"/>
		<param name="recovery_alpha_slow" value="0.0"/>
		<param name="recovery_alpha_fast" value="0.0"/>
	</node>-->
	

	</launch>
